### PR TITLE
script for creating cluster and deploy openebs in azure

### PR DESCRIPTION
**Pull request by nsathyaseelan**
Signed-off-by: nsathyaseelan

**What this PR does / why we need it**:
 This PR will gives the script for creatng aks cluster and setup OpenEBS in Azure

**Special notes for your reviewer**:
While running the above script it gives the below output and it creates the log file in the name of oebs_aks.log

output : 
```
Installing Prerequisites...
Get:1 http://security.ubuntu.com/ubuntu xenial-security InRelease [102 kB]
Hit:2 http://archive.ubuntu.com/ubuntu xenial InRelease
Hit:3 http://archive.ubuntu.com/ubuntu xenial-updates InRelease
Hit:4 http://archive.ubuntu.com/ubuntu xenial-backports InRelease
Fetched 102 kB in 2s (42.9 kB/s)
Reading package lists... Done
Installing Azure CLI
Installed Azure CLI Version: (2.0.29)
Logging into Azure Account
Logged in Account name : admin@*****************
Creating Resource Group
Resource Group created in the name of : aks-openebs-zp3pvh35pf-rg
Creating AKS Cluster.....
AKS cluster created in the name of: aks-openebs-zp3pvh35pf-cluster
Getting credentials to connect aks cluster
Kubectl already Installed, Skipping
Creating Public IP.....
Associating Public IP to the Network Interface...
Installing iSCSI packages
The authenticity of host '13.90.224.116 (13.90.224.116)' can't be established.
ECDSA key fingerprint is SHA256:YYaLJOEPUdRn50hlIs0+0qhOZMu6buaVD4ui5ybVI6c.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added '13.90.224.116' (ECDSA) to the list of known hosts.
Get:1 http://security.debian.org jessie/updates InRelease [63.1 kB]
Ign http://deb.debian.org jessie InRelease
Get:2 http://deb.debian.org jessie-updates InRelease [145 kB]
Get:3 http://deb.debian.org jessie Release.gpg [2434 B]
Get:4 http://deb.debian.org jessie Release [148 kB]
Get:5 http://security.debian.org jessie/updates/main amd64 Packages [643 kB]
Get:6 http://deb.debian.org jessie-updates/main amd64 Packages [23.1 kB]
Get:7 http://deb.debian.org jessie/main amd64 Packages [9064 kB]
Fetched 10.1 MB in 33s (299 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  open-iscsi
0 upgraded, 1 newly installed, 0 to remove and 39 not upgraded.
Need to get 296 kB of archives.
After this operation, 2052 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian/ jessie/main open-iscsi amd64 2.0.873+git0.3b4b4500-8+deb8u2 [296 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 296 kB in 5s (49.7 kB/s)
Selecting previously unselected package open-iscsi.
(Reading database ... 14826 files and directories currently installed.)
Preparing to unpack .../open-iscsi_2.0.873+git0.3b4b4500-8+deb8u2_amd64.deb ...
Unpacking open-iscsi (2.0.873+git0.3b4b4500-8+deb8u2) ...
Processing triggers for systemd (215-17+deb8u7) ...
Running in chroot, ignoring request.
Setting up open-iscsi (2.0.873+git0.3b4b4500-8+deb8u2) ...
invoke-rc.d: policy-rc.d denied execution of start.
update-rc.d: warning: start and stop actions are no longer supported; falling back to defaults
Running in chroot, ignoring request.
update-rc.d: warning: start and stop actions are no longer supported; falling back to defaults
Running in chroot, ignoring request.
Processing triggers for systemd (215-17+deb8u7) ...
Running in chroot, ignoring request.
The authenticity of host '13.82.92.161 (13.82.92.161)' can't be established.
ECDSA key fingerprint is SHA256:3Tfg1N+lngSpzO3zKiE/eib8i0vNNYIKaOacHKW2h9w.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added '13.82.92.161' (ECDSA) to the list of known hosts.
Get:1 http://security.debian.org jessie/updates InRelease [63.1 kB]
Ign http://deb.debian.org jessie InRelease
Get:2 http://deb.debian.org jessie-updates InRelease [145 kB]
Get:3 http://deb.debian.org jessie Release.gpg [2434 B]
Get:4 http://deb.debian.org jessie Release [148 kB]
Get:5 http://security.debian.org jessie/updates/main amd64 Packages [643 kB]
Get:6 http://deb.debian.org jessie-updates/main amd64 Packages [23.1 kB]
Get:7 http://deb.debian.org jessie/main amd64 Packages [9064 kB]
Fetched 10.1 MB in 38s (265 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  open-iscsi
0 upgraded, 1 newly installed, 0 to remove and 39 not upgraded.
Need to get 296 kB of archives.
After this operation, 2052 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian/ jessie/main open-iscsi amd64 2.0.873+git0.3b4b4500-8+deb8u2 [296 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 296 kB in 6s (48.5 kB/s)
Selecting previously unselected package open-iscsi.
(Reading database ... 14826 files and directories currently installed.)
Preparing to unpack .../open-iscsi_2.0.873+git0.3b4b4500-8+deb8u2_amd64.deb ...
Unpacking open-iscsi (2.0.873+git0.3b4b4500-8+deb8u2) ...
Processing triggers for systemd (215-17+deb8u7) ...
Running in chroot, ignoring request.
Setting up open-iscsi (2.0.873+git0.3b4b4500-8+deb8u2) ...
invoke-rc.d: policy-rc.d denied execution of start.
update-rc.d: warning: start and stop actions are no longer supported; falling back to defaults
Running in chroot, ignoring request.
update-rc.d: warning: start and stop actions are no longer supported; falling back to defaults
Running in chroot, ignoring request.
Processing triggers for systemd (215-17+deb8u7) ...
Running in chroot, ignoring request.
Deploying OpenEBS operator...
Deploying OpenEBS Storage classes
Checking Pod status....
kubectl get pods --all-namespaces
NAMESPACE     NAME                                    READY     STATUS    RESTARTS   AGE
kube-system   heapster-2574232661-8bkfd               2/2       Running   0          1m
kube-system   kube-dns-v20-2253765213-2hjkj           3/3       Running   0          3m
kube-system   kube-dns-v20-2253765213-hwnc7           3/3       Running   0          3m
kube-system   kube-proxy-f1hz1                        1/1       Running   0          3m
kube-system   kube-proxy-lvz3b                        1/1       Running   0          3m
kube-system   kube-svc-redirect-92s8s                 1/1       Running   0          3m
kube-system   kube-svc-redirect-mpjpj                 1/1       Running   0          3m
kube-system   kubernetes-dashboard-2898242510-6gxwq   1/1       Running   0          3m
kube-system   tunnelfront-2851197899-tmssw            1/1       Running   0          3m
```

### CODE REVIEW

**Comment by nsathyaseelan** [comment]
@ksatchit @yudaykiran please review this PR

File: e2e/scripts/oebs_aks.sh [hunk 1]
```diff
@@ -0,0 +1,143 @@
+#!/bin/bash
+
+node_count="2" #--- number of nodes for the Azure kubernetes cluster
+node_vm_size="Standard_A2_v2" #--- VM size for the Azure Kubernetes  cluster
```
Comment (ksatchit) [inline e2e/scripts/oebs_aks.sh]: Would recommend to take user inputs with defaults for the node count and vm type.

**Comment by ksatchit** [comment]
Can this be accompanied by a brief note about how to use/run this script, what it does etc.., - preferably in a README?

File: e2e/scripts/oebs_aks.sh (hunk 1 above)
Comment (vharsh) [reply e2e/scripts/oebs_aks.sh]: @nsathyaseelan Refer https://unix.stackexchange.com/a/122848

File: e2e/scripts/Readme.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    # Add /usr/local/bin to PATH
+    PATH="$HOME/bin:$HOME/.local/bin:/usr/local/bin:$PATH"
+    
+    $ source ~/.profile
+```
+**NOTE:** Node count will be the number of nodes, and Node vm size is the size of the virtual machine. For details of node vm size goto the following URL  https://docs.microsoft.com/en-us/azure/cloud-services/cloud-services-sizes-specs
+
+Create a cluster on AKS
+
+Run the  oebs_aks.sh to create AKS cluster and deploy OpenEBS. The above script will installs the Azure CLI, Kubectl,  iSCSI packages on Kubelet container and it deploy OpenEBS operator and OpenEBS storage class files. And It creates the Log file in the name of oebs_aks.log.
```
Comment (yudaykiran) [inline e2e/scripts/Readme.md]: Run the  oebs_aks.sh to create a Kubernetes Cluster on AKS and deploy OpenEBS. 

The script will first install the following prerequisites:
- Azure CLI
- kubectl

Once the Kubernetes Cluster is created, the script goes ahead and installs the ISCSI packages in the kubelet container of all worker nodes.

Finally the script deploys OpenEBS operator and OpenEBS Storage Classes on the cluster.

File: e2e/scripts/Readme.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+```
+    devops@Azure:~$ az group list
+```
+To verify the Created cluster run the following command.
+```
+    devops@Azure:~$ az aks list -g <resource group name>
+```
+To verify the successful deployment of OpenEBS run the following command.
+```
+    devops@Azure:~/openebs/k8s$ kubectl get pods --all-namespaces
```
Comment (yudaykiran) [inline e2e/scripts/Readme.md]: show only OpenEBS pods and change command to `kubectl get pods`

**Pull request merged**